engineering-plans/0001-graphql-query-prefetching.md

```diff
@@ -0,0 +1,397 @@
+# PLAN-0001: GraphQL Query Prefetching
+
+<dl>
+  <dt>Author</dt>
+  <dd>David Lutterkort</dd>
+
+<dt>Implements</dt>
+<dd>No RFC - no user visible changes</dd>
+
+<dt>Engineering Plan pull request</dt>
+<dd><a href="<https://github.com/graphprotocol/rfcs/pull/2>"><https://github.com/graphprotocol/rfcs/pull/2></a></dd>
+
+<dt>Date of submission</dt>
+<dd>2019-11-27</dd>
+
+<dt>Date of approval</dt>
+<dd>2019-12-10</dd>
+
+  <dt>Approved by</dt>
+  <dd>Jannis Pohlmann, Leo Yvens</dd>
+</dl>
+
+This is not really a plan as it was written and discussed before we adopted
+the RFC process, but contains important implementation detail of how we
+process GraphQL queries.
+
+
+## Implementation Details for prefetch queries
+
+
+### Goal
+
+For a GraphQL query of the form
+
+    query {
```

Comment:
Let's do code blocks with

    ```graphql
    ```
and
    ```sql
    ```
rather than indentation. That way we get syntax highlighting on GitHub and in mdBook.

Reply:
Arrgh, I had that in org-mode, but Emacs helpfully stripped it out

Reply:
Yeah, it does that, sadly.